### PR TITLE
remove com.sun.jdi.VMDisconnectedException from reasonable places.

### PR DESCRIPTION
**Pull request by andxu**
https://github.com/Microsoft/java-debug/issues/59

### CODE REVIEW

File: com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/AttachRequestHandler.java [hunk 1]
```diff
@@ -73,7 +75,12 @@ public void handle(Command command, Arguments arguments, Response response, IDeb
                     context.sendEvent(Events.OutputEvent.createConsoleOutput(warnMessage));
                 }
             }
+        } catch (VMDisconnectedException e) {
```
Comment (akaroml) [inline com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/AttachRequestHandler.java]: This exception seems like a common one for all the handlers. Can we handle them at the framework level?

File: com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/Breakpoint.java [hunk 2]
```diff
@@ -66,7 +66,11 @@
     // AutoCloseable
     @Override
     public void close() throws Exception {
-        vm.eventRequestManager().deleteEventRequests(requests());
+        try {
+            vm.eventRequestManager().deleteEventRequests(requests());
+        } catch (VMDisconnectedException ex) {
+            // ignore since removing breakpoints is meanness when JVM is terminated.
```
Comment (testforstephen) [inline com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/Breakpoint.java]: typos. `meaningless`

File: com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/ProtocolServer.java [hunk 3]
```diff
@@ -68,16 +70,32 @@ protected void dispatchRequest(Messages.Request request) {
         this.debugAdapter.dispatchRequest(request).whenComplete((response, ex) -> {
```
Comment (akaroml) [inline com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/ProtocolServer.java]: Use thenAccept.

File: com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/ProtocolServer.java [hunk 4]
```diff
@@ -68,16 +70,32 @@ protected void dispatchRequest(Messages.Request request) {
...
+                                ex.getMessage() != null ? ex.getMessage() : ex.toString()));
+                    }
+                } else {
+                    logger.log(Level.SEVERE, "The request dispatcher should not return null response.");
+                    sendMessage(AdapterUtils.setErrorResponse(response,
+                            ErrorCode.UNKNOWN_FAILURE,
+                            "The request dispatcher should not return null response."));
+                }
+            }
+        }).whenComplete((r, e) -> {
```
Comment (akaroml) [inline com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/ProtocolServer.java]: Use exceptionally().

File: com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/ProtocolServer.java [hunk 5]
```diff
@@ -68,16 +70,32 @@ protected void dispatchRequest(Messages.Request request) {
...
-                sendMessage(AdapterUtils.setErrorResponse(response,
-                        ErrorCode.UNKNOWN_FAILURE,
-                        ex.getMessage() != null ? ex.getMessage() : ex.toString()));
             } else {
-                logger.log(Level.SEVERE, "The request dispatcher should not return null response.");
-                sendMessage(AdapterUtils.setErrorResponse(response,
-                        ErrorCode.UNKNOWN_FAILURE,
-                        "The request dispatcher should not return null response."));
+                response = new Messages.Response(request.seq, request.command);
+                if (ex != null) {
```
Comment (akaroml) [inline com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/ProtocolServer.java]: Put the exception handling logic into the exceptionally() handler.

File: com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/ProtocolServer.java [hunk 6]
```diff
@@ -68,16 +70,32 @@ protected void dispatchRequest(Messages.Request request) {
         this.debugAdapter.dispatchRequest(request).whenComplete((response, ex) -> {
             if (response != null) {
```
Comment (akaroml) [inline com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/ProtocolServer.java]: Don't need to check response for null value when using thenAccept.

**Pull request merged**